File: SchoolAngular/src/app/service/group.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { Group } from "../model/group.model";
import { GroupGeneralView } from "../model/group.general.view.model";

@Injectable()
export class GroupService {

  private basUrl = "http://localhost:8080/group"

  constructor(private httpClient: HttpClient) {

  }

  getGroupList(): Observable<Group[]> {
    return this.httpClient.get<Group[]>(`${this.basUrl}`);
  }

  createGroup(user: Group): Observable<Group> {
    return this.httpClient.post<Group>(`${this.basUrl}`, user);
  }

  getGroupById(id: number): Observable<Group>{
    return this.httpClient.get<Group>(`${this.basUrl}/${id}`);
  }

  updateGroup(id:number, group:Group): Observable<Object>{
    return this.httpClient.put(`${this.basUrl}/${id}`, group);
  }

  deleteGroup(id:number): Observable<Object>{
    return this.httpClient.delete(`${this.basUrl}/${id}`);
  }

  generalView(): Observable<GroupGeneralView[]> {
    return this.httpClient.get<GroupGeneralView[]>(`${this.basUrl}/general-view`);
  }

}

File: springCurseWork/src/main/java/com/example/springcursework/servise/GroupService.java
package com.example.springcursework.servise;

import com.example.springcursework.model.Group;
import com.example.springcursework.model.GroupGeneralView;

import java.util.List;

public interface GroupService {
    public Group insert(Group groupVO);

    public List<Group> findAll();

    public void delete(int id);

    public Group findById(int id);

    public Group updateGroup(int id, Group groupVO);

    public List<GroupGeneralView> generalView();
}
